### ISSUE TITLE
Since version is not properly calculated in properties macros

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#14052]
detected at https://github.com/checkstyle/checkstyle/pull/14049/files#r1398641110

goal is to remove this:
https://github.com/checkstyle/checkstyle/pull/14049/files#diff-582038065de9c242c07931401cfa262fcfd745638bf7c42736dba5793a3294cdR209-R211
and may be whole SINCE_VERSION_FOR_INHERITED_PROPERTY  or heavily make it smaller.

HeaderCheck Should have version 3.0 .
as it existed from 2002 - https://github.com/checkstyle/checkstyle/commits/aca96ae1b180a8036cc23e93840c4161ae8f8927/src/checkstyle/com/puppycrawl/tools/checkstyle/checks/HeaderCheck.java?browsing_rename_history=true&new_path=src/main/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheck.java&original_branch=master

problem is that property was added in 8.3 version but check was added to 8.24.
Before when PropertyMacros calculate since value it should compare it with version of Module and if version of module is above then use it.

We can try to reuse https://stackoverflow.com/a/65077344/1015848 class to compare versions.

Versions from abstract classes should be used with caution https://github.com/checkstyle/checkstyle/pull/14048#issuecomment-1834411335

**Comment by aayushRedHat** [comment]
> @aayushRedHat , can you help with this issue ?

Sure

**Comment by romani** [comment]
I suspecting that whole 
https://github.com/checkstyle/checkstyle/blob/4f0aeba32b5d11a9d3f0334f94c077e569f29097/src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java#L166-L176
is present only due to this defect.

**Comment by rnveach** [comment]
> is present only due to this defect.

Even so, this workaround must stay because even this method can fail.

**Comment by romani** [comment]
yes, but we can significantly shrink content of this collection.

>Map.entry("MissingDeprecatedCheck.violateExecutionOnNonTightHtml", V824)

is same problem as issue description, we should be able to remove it in scope of this issue.

**Comment by relentless-pursuit** [comment]
Hi @Romani, any progress here? Are we still facing the issue?

**Comment by romani** [comment]
Still a problem, load of PRs is too big to let me do any coding, please help to fix this issue.

**Comment by relentless-pursuit** [comment]
> Still a problem, load of PRs is too big to let me do any coding, please help to fix this issue.

Sure, will start shortly, after making some decent progress on pending PRs.

**Comment by relentless-pursuit** [comment]
I won't be able to continue my work on this PR, as I am busy with other open PRs, and following up with one of the GSOC projects. I will surely combeback to this once later, if not been taken by anybody else.

**Comment by smita1078** [comment]
@romani I am on it, just to get a clean understanding... Correct me if I am wrong
for eg: in 
`Map.entry("MissingDeprecatedCheck.violateExecutionOnNonTightHtml", V824)` what I understand is that property was introduced in  8.24 but module was added in 8.3, for now it is showing 8.24 but it should rather show 8.3 as module version is above?

**Comment by romani** [comment]
property appear at updaate https://github.com/checkstyle/checkstyle/commit/2f8bcacc1e22b2ad0e249a0f942f799e47f23d0a#diff-e443a0018a3dae545ad69f558f511366d5d8d9ea032edc26ea719d24bbb0cdedR68

linked to 
https://github.com/checkstyle/checkstyle/issues/4983 ,released in 8.24.

property appear in 8.24 version for users.

Goal of issue is to somehow avoid this hardcoded map of versions and use data from sources.
But in this case, Class since is "5.0", abstract class property is "8.3", and there is no place to have "8.24" as we updated inheritance for Check (invisible operation for user). But it affects visible properties.

What we can try to do is:
- make permanent map of such overrides of versions for such cases and focus on less problematic cases for now.
- override property to put in Check to put propert javadoc on it.

**Comment by SteLeo1602** [comment]
This is issue is finished by my PRs specified above
